sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -124,11 +153,24 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
     val hiveVersion = externalCatalog.unwrapped.asInstanceOf[HiveExternalCatalog].client.version
     val stagingDir = hadoopConf.get("hive.exec.stagingdir", ".hive-staging")
     val scratchDir = hadoopConf.get("hive.exec.scratchdir", "/tmp/hive")
+    logDebug(s"path '${path.toString}', staging dir '$stagingDir', " +
+      s"scratch dir '$scratchDir' are used")
 
     if (hiveVersionsUsingOldExternalTempPath.contains(hiveVersion)) {
       oldVersionExternalTempPath(path, hadoopConf, scratchDir)
     } else if (hiveVersionsUsingNewExternalTempPath.contains(hiveVersion)) {
```

Comment:
Why this feature only applies on `hiveVersionsUsingNewExternalTempPath` cases? How about `hiveVersionsUsingOldExternalTempPath`?

Reply:
It is because old versions do not support data copy between different type of DFSs.

Reply:
You mean old Hive versions doesn't support it? If so, I think we should at least document it clearly.

Reply:
Got it. I added the descroption "This option is supported in Hive 2.0 or later." in SQLConf.scala.
https://github.com/apache/spark/pull/27690/files#diff-9a6b543db706f1a90f790783d6930a13R849